coordinator-common/src/main/java/org/apache/kafka/coordinator/common/runtime/CoordinatorRuntime.java

```diff
@@ -2127,6 +2143,10 @@ private CoordinatorRuntime(
         this.compression = compression;
         this.appendLingerMs = appendLingerMs;
         this.executorService = executorService;
+        this.appendMaxBufferSizeSupplier = appendMaxBufferSizeSupplier;
+        this.runtimeMetrics.registerAppendBufferSizeGauge(
+            () -> coordinators.values().stream().mapToLong(c -> c.cachedBufferSize).sum()
```

Comment:
Using `Supplier<Long>` necessiates synchronization, as the `cachedBufferSize` value is accessed by different threads